Sources/Kitura/contentType/MediaType.swift

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright IBM Corporation 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import KituraNet
+
+/**
+ The media type (formerly known as MIME type) is a standardized way to indicate the nature and format of a document.
+ This struct consists of a catagorical `topLevelType` and specific `subtype` seperated by a "/" (e.g. "text/plain").
+ In HTTP, The media type is sent as the first section of the "Content-Type" header and is case insensitive.
+ ### Usage Example: ###
+ ```swift
+ let mediaType = MediaType(type: .application, subtype: "json")
+ print(mediaType.description)
+ // Prints ("application/json")
+ ```
+ */
+public struct MediaType: CustomStringConvertible, Equatable, Hashable {
```

Comment:
Why is this `public`? Seems like something we're going to use internally inside Kitura but not expose to users.